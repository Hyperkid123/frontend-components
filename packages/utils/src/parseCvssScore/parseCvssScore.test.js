import React from 'react';
import parseCvssScore from './parseCvssScore';
import { render } from '@testing-library/react';

describe('parseCvssScore', () => {
  describe('no cvss', () => {
    const result = parseCvssScore();
    const { container } = render(result);
    it('render', () => {
      expect(container).toMatchSnapshot();
    });
  });

  it('cvssV3', () => {
    const result = parseCvssScore(1, 2);
    const { container } = render(<React.Fragment>{result}</React.Fragment>);
    expect(container).toMatchSnapshot();
  });

  describe('cvssV2', () => {
    const result = parseCvssScore(2);
    const { container } = render(result);
    it('should render v2 cvss', () => {
      expect(container).toMatchSnapshot();
    });
  });

  describe('cvssV2 with labels', () => {
    const result = parseCvssScore(2, 0, true);
    const { container } = render(result);
    it('should render v2 cvss with label', () => {
      expect(container).toMatchSnapshot();
    });

    it('should render correct tooltip content', () => {
      expect(container).toMatchSnapshot();
    });
  });
});
