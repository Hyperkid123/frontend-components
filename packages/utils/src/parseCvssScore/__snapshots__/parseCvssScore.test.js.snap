// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseCvssScore cvssV2 should render v2 cvss 1`] = `<div />`;

exports[`parseCvssScore cvssV2 with labels should render correct tooltip content 1`] = `<div />`;

exports[`parseCvssScore cvssV2 with labels should render v2 cvss with label 1`] = `<div />`;

exports[`parseCvssScore cvssV3 1`] = `
<div>
  2.0
</div>
`;

exports[`parseCvssScore no cvss render 1`] = `
<div>
  <div
    style="display: contents;"
  >
    <span>
      N/A
    </span>
  </div>
</div>
`;
